exercise-testing-basics.js/test/students-submissions/yoav-ben-zvi.test.js

```diff
@@ -0,0 +1,176 @@
+		expect(true).toBe(true);
+		// ✅ TASK: Wrap this simple test above with a 'describe' statement
+	});
+});
+
+describe("Writing another exampe to excercise writing assertions", () => {
+	test("Another example test", () => {
+		expect(true).toBe(true);
+		expect(42).not.toBe(52);
+		expect(4 !== 3).toBeTruthy();
+		expect(-200).toBeLessThan(1);
+		expect({ name: "Yoav" }).toEqual({ name: "Yoav" });
+		expect(-4 + 7).not.toBeNegative();
+		expect.assertions(8);
+		expect(new Number()).toBeInstanceOf(Number);
+		expect([][2]).toBeUndefined();
+		// ✅ TASK: Add your first test 🎉. This test should not test real code, just choose a title and put one hard-coded assertion (expect)
+	});
+});
+
+const userExample = {
+	name: "Kent",
+	familyName: "Beck",
+	zipCode: "32486-01",
+	address: "Moonlight road 181, Alaska",
+};
+
+describe("Testing the 'validateUser' method of 'userService'", () => {
+	test("When no name provided, then validation should fail with false response", () => {
+		//Arrange
+		const userWithNoName = {
+			familyName: "Beck",
+			zipCode: "32486-01",
+			address: "Moonlight road 181, Alaska",
+		};
+		const userServiceUnderTest = new usersService();
+
+		//Act
+		const recievedResult = userServiceUnderTest.validateUser(userWithNoName);
+
+		//Assert
+		expect(recievedResult.succeeded).toBe(false);
+		// ✅ TASK: Test the 'validateUser' method of the 'usersService': Ensure that when no 'name' property is provided,
+		//    the response 'succeeded' property is false. Remember the fail-first principle, ensure the test fail when appropriate
+		// ✅ TASK: Use the AAA pattern in the test you just coded above ☝🏻
+		// 💡 TIP: Put 3 sections within the test (appear below). In each one of them, place the appropriate parts
+	});
+});
+
+describe("Testing the 'validateUser' method of 'usersService'", () => {
+	test("Testing that the response is satisfactory", () => {
+		//Arrange
+		const fullUserExample = {
+			name: "Kent",
+			familyName: "Beck",
+			zipCode: "32486-01",
+			address: "Moonlight road 181, Alaska",
+		};
+		const userServiceUnderTest = new usersService();
+
+		//Act
+		const recievedResult = userServiceUnderTest.validateUser(fullUserExample);
+
+		//Assert
+		expect(recievedResult).toEqual({
+			succeeded: true,
+			reasons: [],
+		});
+		// ✅ TASK: Create another test against the 'validateUser' method of the 'usersService':
+		//    Provide all the mandatory field and ensure that the response is satisfactory (as expected)
+	});
+
+	test("Testing the 'reasons' array against missing properties in the 'validateUser' method of 'usersService'", () => {
+		//Arrange
+		const UserExampleWithMissingProperties = {
+			name: "Kent",
+			address: "Moonlight road 181, Alaska",
+		};
+		const userServiceUnderTest = new usersService();
+
+		//Act
+		const recievedResult = userServiceUnderTest.validateUser(
+			UserExampleWithMissingProperties
+		);
+
+		//Assert
+		expect(recievedResult.reasons).toIncludeAllMembers(["no-name"]);
```

Comment:
@Yoavbenzvi What about the address failure reason? See how your test pass, but won't discover a bug in case address validation forgets to add a reason to the array